vaas/vaas/cluster/models.py

```diff
@@ -138,16 +139,24 @@ def mappings(self) -> Set[str]:
     def labels(self, mappings: Union[List, Set]) -> None:
         self._mappings, self.mappings_list = self._prepare_set_and_json(mappings)
 
-    def mapped_domains(self, cluster: LogicalCluster) -> List[str]:
+    class Replacement:
```

Comment:
I don't know if a nested class is what we see in our code base.
Please consider moving out of this class.